### DISCUSSION TITLE
Flask 10.28 has been released! Look here for breaking changes.

### DISCUSSION THREAD

**Discussion by FrederikBolding** [MetaMask/snaps#1348] [Announcements]
Flask `10.28.2` has been released to Chrome and Firefox. See below for breaking changes:

- Changed checksum calculation algorithm
  - Every snap needs to have its checksum/shasum regenerated and have a new version published
  - To use the new checksum algorithm please update your CLI `@metamask/snaps-cli` to at least version `0.30.0`
- Removed `snap_confirm`
  - Please use `snap_dialog` instead (see [documentation](https://docs.metamask.io/snaps/reference/rpc-api#snap_dialog))
- `WebAssembly` is not available by default anymore
   - Snaps that want to use it must request the [endowment:webassembly](https://docs.metamask.io/snaps/reference/permissions#endowmentwebassembly) permission

**Comment by samtay** [comment]
I have an `ethers` dependency specified in my snap. With this new upgrade, `import { ethers } from 'ethers';` in my `packages/snap/src/index.ts` causes an error whenever I try to install the snap:
```
An error happened: Error while running snap 'local:http://localhost:8080': Illegal invocation
```
If I comment out the import in that file, the app installs fine. Any ideas?

**Reply by FrederikBolding** [reply]
Hey there!

This is a known bug with code using `atob` and `btoa`.

If you are comfortable with it you can build the MetaMask extension from source (https://github.com/MetaMask/metamask-extension) using `yarn dist --build-type=flask`. Let me know if that doesn't work for you. If you aren't comfortable with that, you'll unfortunately have to wait for the next Flask release.

**Reply by samtay** [reply]
Oh that's interesting. I want people to be able to interact with the dapp without having to build extensions from source, so I'll have to wait until next release. In the meantime I should be able to move my ethers usage out to the dapp side; it will make a few things awkward, but not the end of the world. Thanks for your quick response!

**Reply by luketchang** [reply]
When can we expect the next Flask release for the `atob`/`btoa` patch?

**Reply by FrederikBolding** [reply]
@luketchang Some time next week most likely. Will publish it as soon as I can!

**Reply by panieldark** [reply]
Please do! Super unfortunate interaction right now with `Illegal invocation`, would very much appreciate having the canary version back up and running as expected, for everyone 😄

**Reply by FrederikBolding** [reply]
@panieldark @samtay @luketchang Flask `10.29.0` is out now! Please let me know if you experience more issues!

**Reply by samtay** [reply]
There's still an issue. If the snap depends on `ethers: 6.3.0` and you `import { toBeHex } from 'ethers';`, the build fails:

```
❯ yarn build
Build error: Parsing file /home/sam/code/snap/node_modules/ethers/lib.commonjs/abi/abi-coder.js: Unexpected token (112:4)
Error: Parsing file /home/sam/code/snap/node_modules/ethers/lib.commonjs/abi/abi-coder.js: Unexpected token (112:4)
    at Deps.parseDeps (/home/sam/code/snap/node_modules/module-deps/index.js:519:15)
    at getDeps (/home/sam/code/snap/node_modules/module-deps/index.js:447:44)
    at /home/sam/code/snap/node_modules/module-deps/index.js:430:38
    at ConcatStream.<anonymous> (/home/sam/code/snap/node_modules/concat-stream/index.js:37:43)
    at ConcatStream.emit (node:events:525:35)
    at finishMaybe (/home/sam/code/snap/node_modules/readable-stream/lib/_stream_writable.js:630:14)
    at endWritable (/home/sam/code/snap/node_modules/readable-stream/lib/_stream_writable.js:638:3)
    at Writable.end (/home/sam/code/snap/node_modules/readable-stream/lib/_stream_writable.js:594:41)
    at DuplexWrapper.onend (/home/sam/code/snap/node_modules/readable-stream/lib/_stream_readable.js:577:10)
    at Object.onceWrapper (node:events:627:28)
    at DuplexWrapper.emit (node:events:525:35)
    at endReadableNT (/home/sam/code/snap/node_modules/readable-stream/lib/_stream_readable.js:1010:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  stream: Labeled {
    _readableState: ReadableState {
      objectMode: true,
      highWaterMark: 16,
      buffer: BufferList { length: 0 },
      length: 0,
      pipes: [Labeled],
      pipesCount: 1,
      flowing: true,
      ended: false,
      endEmitted: false,
      reading: true,
      sync: false,
      needReadable: true,
      emittedReadable: false,
      readableListening: false,
      resumeScheduled: false,
      destroyed: false,
      defaultEncoding: 'utf8',
      awaitDrain: 0,
      readingMore: false,
      decoder: null,
      encoding: null
    },
    readable: true,
    _events: [Object: null prototype] {
      end: [Array],
      error: [Function (anonymous)],
      data: [Function: ondata],
      _mutate: [Function]
    },
    _eventsCount: 4,
    _maxListeners: undefined,
    _writableState: WritableState {
      objectMode: true,
      highWaterMark: 16,
      finalCalled: false,
      needDrain: false,
      ending: true,
      ended: true,
      finished: true,
      destroyed: false,
      decodeStrings: true,
      defaultEncoding: 'utf8',
      length: 0,
      writing: false,
      corked: 0,
      sync: false,
      bufferProcessing: false,
      onwrite: [Function (anonymous)],
      writecb: null,
      writelen: 0,
      bufferedRequest: null,
      lastBufferedRequest: null,
      pendingcb: 0,
      prefinished: true,
      errorEmitted: false,
      bufferedRequestCount: 0,
      corkedRequestsFree: [CorkedRequest]
    },
    writable: false,
    allowHalfOpen: true,
    _options: { objectMode: true },
    _wrapOptions: { objectMode: true },
    _streams: [ [Deps] ],
    length: 1,
    label: 'deps',
    [Symbol(kCapture)]: false
  }
}
```

The unexpected token is I think this `#` in `abi-coder.js`:

```js
class AbiCoder {
    #getCoder(param) {
        if (param.isArray()) {
```

**Reply by samtay** [reply]
I suppose this is a separate issue. Importing `import { BigNumber, BigNumberish } from 'ethers';` from `ethers: 5.7.2` works fine.

**Reply by ziad-saab** [reply]
I was able to reproduce the issue. We'll look into it.

**Reply by FrederikBolding** [reply]
@samtay Please try the latest `@metamask/snaps-cli`: `0.33.0-flask.1` - that should fix your issue. Let me know if it doesn't!

**Reply by samtay** [reply]
> @samtay Please try the latest `@metamask/snaps-cli`: `0.33.0-flask.1` - that should fix your issue. Let me know if it doesn't!

Hm that didn't work for me :confused:. I've got `@metamask/snaps-cli`, `@metamask/snaps-types`, `@metamask/snaps-ui` all set to `0.33.0-flask.1`. If I `import { ethers } from 'ethers';` I get the same error as above (on `abi-coder.js`). If I pick out `import { toBeHex, formatEther } from 'ethers';` then I get a very similar error but from a different file. Looking at the line in question, it still appears to be an issue parsing the private symbol `#`.

**Comment by panieldark** [comment]
@FrederikBolding Cheers guys, fixed 👍
